includes/create-theme/theme-patterns.php

```diff
@@ -47,4 +71,111 @@ public static function create_pattern_link( $attributes ) {
+		foreach ( array_merge( $patterns, $templates, $template_parts ) as $file ) {
+			$file_content = file_get_contents( $file );
+			$file_content = str_replace( 'wp:block {"ref":' . $pattern->id . '}', 'wp:pattern {"slug":"' . $pattern->slug . '"}', $file_content );
+			file_put_contents( $file, $file_content );
+		}
+	}
+
+	public static function prepare_pattern_for_export( $pattern, $options = null ) {
+		if ( ! $options ) {
+			$options = array(
+				'localizeText'   => false,
+				'removeNavRefs'  => true,
+				'localizeImages' => true,
+			);
+		}
+
+		$pattern = CBT_Theme_Templates::eliminate_environment_specific_content( $pattern, $options );
+
+		if ( array_key_exists( 'localizeText', $options ) && $options['localizeText'] ) {
+			$pattern = CBT_Theme_Templates::escape_text_in_template( $pattern );
+		}
+
+		if ( array_key_exists( 'localizeImages', $options ) && $options['localizeImages'] ) {
+			$pattern = CBT_Theme_Media::make_template_images_local( $pattern );
+
+			// Write the media assets if there are any
+			if ( $pattern->media ) {
+				CBT_Theme_Media::add_media_to_local( $pattern->media );
+			}
+		}
+
+		return $pattern;
+	}
+
+	/**
+	 * Copy the local patterns as well as any media to the theme filesystem.
+	 */
+	public static function add_patterns_to_theme( $options = null ) {
+		$base_dir     = get_stylesheet_directory();
+		$patterns_dir = $base_dir . DIRECTORY_SEPARATOR . 'patterns';
+
+		$pattern_query = new WP_Query(
+			array(
+				'post_type'      => 'wp_block',
+				'posts_per_page' => -1,
+			)
+		);
+
+		if ( $pattern_query->have_posts() ) {
+			// If there is no patterns folder, create it.
+			if ( ! is_dir( $patterns_dir ) ) {
+				wp_mkdir_p( $patterns_dir );
+			}
+
+			foreach ( $pattern_query->posts as $pattern ) {
+				$pattern        = self::pattern_from_wp_block( $pattern );
+				$pattern        = self::prepare_pattern_for_export( $pattern, $options );
+				$pattern_exists = false;
+
+				// Check pattern is synced before adding to theme.
+				if ( 'unsynced' !== $pattern->sync_status ) {
+					// Check pattern name doesn't already exist before creating the file.
+					$existing_patterns = glob( $patterns_dir . DIRECTORY_SEPARATOR . '*.php' );
+					foreach ( $existing_patterns as $existing_pattern ) {
+						if ( strpos( $existing_pattern, 'pattern-' . $pattern->name . '.php' ) !== false ) {
+							$pattern_exists = true;
+						}
+					}
+
+					if ( $pattern_exists ) {
+						return new WP_Error(
+							'pattern_already_exists',
+							sprintf(
+								/* Translators: Pattern name. */
+								__(
+									'A pattern with this name already exists: "%s".',
+									'create-block-theme'
+								),
+								$pattern->name
+							)
+						);
+					}
+
+					// Create the pattern file.
+					$pattern_file = $patterns_dir . 'pattern-' . $pattern->name . '.php';
+					file_put_contents(
+						$patterns_dir . DIRECTORY_SEPARATOR . 'pattern-' . $pattern->name . '.php',
```

Comment:
Do you think the `pattern-` prefix is necessary?

```suggestion
						$patterns_dir . DIRECTORY_SEPARATOR . 'pattern-' . $pattern->name . '.php',
```

Reply:
This threw me off. I could not find my patterns and thought it was not working ...until I saw this hint:

> Do you think the pattern- prefix is necessary?

Reply:
Yeah same! Especially if the theme already has lots of patterns it was hard to spot.

Reply:
I think I expected the file name to be the kebab-cased pattern name.

Reply:
I'm not sure why I added the prefix, haha 🙈 I'll remove it for now, I can't see what value it's adding. If I remember I'll let you know!

Reply:
Removed the prefix here: https://github.com/WordPress/create-block-theme/pull/675/commits/f017b9485cfddac0bc0f0f5cf2f5312304f42c4e.